persistent/Database/Persist/Sql/Orphan/PersistUnique.hs

```diff
@@ -18,43 +21,75 @@ import qualified Data.Conduit.List as CL
 import Control.Monad.Trans.Reader (ask, withReaderT)
 import Control.Monad (when, liftM)
 
-defaultUpsert :: (MonadIO m, PersistEntity record, PersistUniqueWrite backend
-                 , PersistEntityBackend record ~ BaseBackend backend) 
-                => record -> [Update record] -> ReaderT backend m (Entity record)
+defaultUpsert
+    :: (MonadIO m
+       ,PersistEntity record
+       ,PersistUniqueWrite backend
+       ,PersistEntityBackend record ~ BaseBackend backend)
+    => record -> [Update record] -> ReaderT backend m (Entity record)
 defaultUpsert record updates = do
-  uniqueKey <- onlyUnique record
-  upsertBy uniqueKey record updates
+    uniqueKey <- onlyUnique record
+    upsertBy uniqueKey record updates
 
-instance PersistUniqueWrite SqlBackend where
+escape :: DBName -> T.Text
+escape (DBName s) = T.pack $ '"' : go (T.unpack s) ++ "\""
+  where
+    go "" = ""
+    go ('"':xs) = "\"\"" ++ go xs
+    go (x:xs) = x : go xs
```

Comment:
I think hlint would say to "avoid explicit recursion" here. I'm bad about "pointless" code golf, but since I'm never getting that time back anyway:

```haskell

escape :: DBName -> T.Text
escape = quote . T.foldr esc "" . unDBName
  where
    quote = T.cons '"' . flip T.snoc '"'
    esc '"' = mappend "\"\""
    esc c   = T.cons c
```

Feel free to ignore. :cat2: